int64 x
int64 y
---
int64 x_goal
int64 y_goal
